test/CarouselSpec.tsx

```diff
@@ -0,0 +1,1025 @@
+      });
+    });
+  });
+
+  describe('mouse events', () => {
+    let clock: sinon.SinonFakeTimers;
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    it('should handle a defined mouse over event', () => {
+      const onMouseOverSpy = sinon.spy();
+
+      const { getByTestId } = render(
+        <Carousel
+          activeIndex={1}
+          onMouseOver={onMouseOverSpy}
+          data-testid="test"
+        >
+          {items}
+        </Carousel>,
+      );
+
+      fireEvent.mouseOver(getByTestId('test'));
+      clock.tick(1500);
+      onMouseOverSpy.should.have.been.calledOnce;
+    });
+
+    it('should handle a defined mouse out event', () => {
+      const onMouseOutSpy = sinon.spy();
+
+      const { getByTestId } = render(
+        <Carousel activeIndex={1} onMouseOut={onMouseOutSpy} data-testid="test">
+          {items}
+        </Carousel>,
+      );
+
+      fireEvent.mouseOut(getByTestId('test'));
+      clock.tick(50);
+      onMouseOutSpy.should.have.been.calledOnce;
+    });
+  });
+
+  describe('touch events', () => {
+    let clock: sinon.SinonFakeTimers,
+      renderResult: RenderResult,
+      carousel: HTMLElement,
+      onSelectSpy: sinon.SinonSpy,
+      onTouchStartSpy: sinon.SinonSpy,
+      onTouchMoveSpy: sinon.SinonSpy,
+      onTouchEndSpy: sinon.SinonSpy;
+
+    beforeEach(() => {
+      onSelectSpy = sinon.spy();
+      onTouchStartSpy = sinon.spy();
+      onTouchMoveSpy = sinon.spy();
+      onTouchEndSpy = sinon.spy();
+
+      renderResult = render(
+        <Carousel
+          activeIndex={1}
+          interval={null}
+          onSelect={onSelectSpy}
+          onTouchStart={onTouchStartSpy}
+          onTouchMove={onTouchMoveSpy}
+          onTouchEnd={onTouchEndSpy}
+          touch
+          data-testid="carousel-test"
+        >
+          {items}
+        </Carousel>,
+      );
+
+      carousel = renderResult.getByTestId('carousel-test');
+
+      clock = sinon.useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    function generateTouchEvents(params: {
+      target: HTMLElement;
+      startX: number;
+      endX: number;
+    }): void {
+      const { target, startX, endX } = params;
+
+      /**
+       * Below code is not working on Firefox due to Touch is not defined error.
+       * Maybe related to {@link https://bugzilla.mozilla.org/show_bug.cgi?id=1693172}?
+       *
+       * To avoid issue we are going to use {@link import('react-dom').Simulate} (used by enzyme internally)
+       */
+
+      // fireEvent.touchStart(target, {
+      //   touches: [new Touch({ identifier: 1, target, clientX: startX })],
+      // });
+      // fireEvent.touchMove(target, {
+      //   touches: [new Touch({ identifier: 1, target, clientX: endX })],
+      // });
+      // fireEvent.touchEnd(target);
+
+      Simulate.touchStart(target, {
```

Comment:
On FireFox there are errors due to missing Touch class. (See [build log for details](https://github.com/react-bootstrap/react-bootstrap/runs/4940744877?check_suite_focus=true)).

In order to fix the issue I dig up how enzyme works and I see that they use `Simulate` from `react-dom/test-utils`.
Using this utility seems to resolve the issue ([Build log for details](https://github.com/react-bootstrap/react-bootstrap/runs/4943579729?check_suite_focus=true), Chrome builds stopped before running test so I think that everything should be fine).

Let me know if this solution is feasible for you.

Thanks!

Reply:
Re-execute of the action after force push the same code produces no error. 
Ready for review / merge.